terraform/main.tf

```diff
@@ -0,0 +1,124 @@
+}
+
+terraform {
+  backend "s3" {
+  }
+}
+
+# Configure the remote state data source to acquire configuration
+# created through the code in ch-service-terraform/aws-mm-networks.
+data "terraform_remote_state" "networks" {
+  backend = "s3"
+  config = {
+    bucket = var.remote_state_bucket
+    key    = "${var.state_prefix}/${var.deploy_to}/${var.deploy_to}.tfstate"
+    region = var.aws_region
+  }
+}
+locals {
+  vpc_id            = data.terraform_remote_state.networks.outputs.vpc_id
+  application_ids   = data.terraform_remote_state.networks.outputs.application_ids
+  application_cidrs = data.terraform_remote_state.networks.outputs.application_cidrs
+}
+
+# Configure the remote state data source to acquire configuration created
+# through the code in aws-common-infrastructure-terraform/groups/networking.
+data "terraform_remote_state" "networks_common_infra" {
+  backend = "s3"
+  config = {
+    bucket = var.aws_bucket
+    key    = "aws-common-infrastructure-terraform/common/networking.tfstate"
+    region = var.aws_region
+  }
+}
+locals {
+  internal_cidrs = values(data.terraform_remote_state.networks_common_infra.outputs.internal_cidrs)
+  vpn_cidrs      = values(data.terraform_remote_state.networks_common_infra.outputs.vpn_cidrs)
+}
+
+# Configure the remote state data source to acquire configuration
+# created through the code in the services-stack-configs stack in the
+# aws-common-infrastructure-terraform repo.
+data "terraform_remote_state" "services-stack-configs" {
+  backend = "s3"
+  config = {
+    bucket = var.aws_bucket # aws-common-infrastructure-terraform repo uses the same remote state bucket
+    key    = "aws-common-infrastructure-terraform/common/services-stack-configs.tfstate"
+    region = var.aws_region
+  }
+}
+
+provider "vault" {
+  auth_login {
+    path = "auth/userpass/login/${var.vault_username}"
+    parameters = {
+      password = var.vault_password
+    }
+  }
+}
+
+data "vault_generic_secret" "secrets" {
+  for_each = toset(var.vault_secrets)
+  path = "applications/${var.aws_profile}/${var.environment}/${local.stack_name}/${each.value}"
+}
+
+locals {
+  # stack name is hardcoded here in main.tf for this stack. It should not be overridden per env
+  stack_name  = "test-data-generator"
+  name_prefix = "${local.stack_name}-${var.environment}"
+}
+
+module "ecs-cluster" {
+  source = "git::git@github.com:companieshouse/terraform-library-ecs-cluster.git?ref=1.1.0"
+
+  stack_name  = local.stack_name
+  name_prefix = local.name_prefix
+  environment = var.environment
+
+  vpc_id                     = local.vpc_id
+  ec2_key_pair_name          = var.ec2_key_pair_name
+  ec2_instance_type          = var.ec2_instance_type
+  ec2_image_id               = var.ec2_image_id
+  asg_max_instance_count     = var.asg_max_instance_count
+  asg_min_instance_count     = var.asg_min_instance_count
+  asg_desired_instance_count = var.asg_desired_instance_count
+  application_subnet_ids     = local.application_ids
```

Comment:
I think we should look at renaming `application_subnet_ids` as `application` doesn't really mean anything

Reply:
Yeah, I could do. Those are external names though...
**_application_subnet_ids_** is the local variable in the cluster library [here](https://github.com/companieshouse/terraform-library-ecs-cluster).
**_application_ids_** is defined in ch-service-terraform [here](https://github.com/companieshouse/ch-service-terraform/blob/master/aws-mm-networks/main.tf) and is in the remote state. I could change the local variable name, but maybe best to keep them consistent?
What do you think?

Reply:
Renamed the `application_subnet_ids` variable here to drop the 'application' prefix. There's now another PR that goes with this for the same change in the terraform-library-ecs-cluster repo [here](https://github.com/companieshouse/terraform-library-ecs-cluster/pull/4).